### ISSUE TITLE
Show specific number of pets/mounts in "Show More"

### ISSUE THREAD

**Issue by lemoness** [HabitRPG/habitica#9786]
Right now, it's not always obvious to users that the Show More dropdowns contain the rest of their pets and mounts. For that reason, we'd like to show the total number of pets and mounts in a given section, as follows:

![new stable](https://user-images.githubusercontent.com/3494081/34803152-92ff3d98-f626-11e7-85ee-e45f703fe9aa.png)

We may also want to apply this to other "Show More" buttons around the site.

**Comment by mandyzhou** [comment]
Hmm, I find the proposed wording a bit confusing -- just from the screenshot, it's not clear whether the 23, for example,  refers to the total number of pets owned in the Standard category, or the number of owned Standard pets that are currently hidden, and the proposed wording does not convey that the Show buttons also reveal paw prints for pets and mounts not currently in users' stables. 

Maybe badges next to the "Standard", "Magic Potion", and other headings, similar to the "12/90" badge currently next to "Pets", would be more helpful for conveying totals? And maybe something like "Show Rest of Standard Pets" or "Show __ More Standard Pets" would be clearer for the "Show More" buttons?

**Comment by Alys** [comment]
I agree that the "Show 23 Standard Pets" wording doesn't make it clear whether there's 23 more or 23 in total.

**Comment by lemoness** [comment]
What about "Show All 23 Standard Pets"?

**Comment by mandyzhou** [comment]
Hmm, maybe I'm overthinking it, but that could be interpreted as saying that there are 23 standard pets it is possible to own. Maybe something like "Show All Standard Pets Owned (23)"?

**Comment by lemoness** [comment]
That seems a bit long to me. "Show 23 Owned Standard Pets" is a little
shorter, but still clunky. Perhaps we should circle back to the design of
it? That being said, it will make it take longer. Perhaps the "Show 23
Standard Pets" could be added as a stop-gap, and hopefully that will reduce
some of the questions about where the pets have gone while we revisit this.

**Comment by seekheart** [comment]
Could it be something like "Show All" and have an indicator of how many pets owned in a chip?

**Comment by ESWAT** [comment]
What about showing the number of owned pets/mounts in the heading?

![screen shot 2018-03-19 at 23 53 09](https://user-images.githubusercontent.com/65468/37635210-c021f5ec-2bd0-11e8-93e3-b79c897f47ff.png)

**Comment by akmaekki** [comment]
Hi, I would like to work on this bug / feature.

**Comment by Alys** [comment]
@akmaekki Fantastic, thanks! Post here if you have any questions.

**Comment by akmaekki** [comment]
I have a question, referring to the screenshot in the first post.

- 12/90 means that there are 90 pets and the user already owns 12, am I right?
- What do we want to display additionally?

**Comment by Alys** [comment]
@akmaekki 

> 12/90 means that there are 90 pets and the user already owns 12, am I right?

That's right, and that's already working on the website.

> What do we want to display additionally?

The "Show More" buttons on the Inventory > Stable page need to be changed as the fix for this issue. If you look on the habitica.com website now, you'll see they say just "Show More". The screenshot in the top post on this issue shows what they should say (i.e., add the number of pets owned in each section).

Note that the screenshot is just a mock-up and doesn't contain accurate data (i.e., the number in the "12/90" text doesn't match the number shown in the button).

**Comment by akmaekki** [comment]
Thank you Alys. Now I got it and will orient toward the screenshot. Previously it was not clear to me, if you agreed upon a design solution.

**Comment by akmaekki** [comment]
I now have a "working version". The screenshot displays how it looks right now. Is this ok in principle? 

Another question arises => if the user has 0 or 1 pet, the string should display singular "standard pet". If the user has more than 1 pet, it should display "standard pets", am I right? If so and plural is important, I would probably introduce a new variable to pets.json. But this would require all of these files to be modified, which is quite an effort and I am in doubt if such a small feature warrants such an effort :-)

Another question, should the section in toggled state, when the button shows "Show less" also display the amount of owned pets? Because I think that just displaying "Show less" is fine?!

![image](https://user-images.githubusercontent.com/17971153/64913000-3ba01c80-d738-11e9-9321-579cec2d7ec0.png)

**Comment by akmaekki** [comment]
Hi, is there anyone who can help me with this?

**Comment by Alys** [comment]
@akmaekki I'm sorry we haven't replied before now!

Firstly as a general note, when new strings do need to be added to the locales files, you have to add them to only the files in the "en" directory. From there, they are uploaded to Weblate (the website that contributors do translations in) and they're automatically copied to all the other language files. Thus it's easy for a programmer to add a new string, and not much burden for the translators either. If you'd like more information, see the _README_FIRST.md file in the locales/en directory.

> I now have a "working version". The screenshot displays how it looks right now. Is this ok in principle?

I'll leave that for the staff to give a definitive answer but it certainly looks like you're on the right track. I think it's worth you submitting a pull request now and then any specific feedback could be made once we've seen your code changes.

> Another question arises => if the user has 0 or 1 pet, the string should display singular "standard pet". If the user has more than 1 pet, it should display "standard pets", am I right? If so and plural is important, I would probably introduce a new variable to pets.json.

Plurals can be troublesome in other languages. Even if you have separate strings for 1 and more than 1, that's not enough for some languages. For example, for Russian there are different plural endings when the number of items ends in 2, 3, or 4, compared to numbers ending with other digits, so just two strings wouldn't be enough to produce perfect translations in every case.

My personal feeling is that since players pretty quickly get more than 1 pet, it's not worth having a separate string for "standard pet". However if the staff say otherwise, listen to them not me. :)

> Another question, should the section in toggled state, when the button shows "Show less" also display the amount of owned pets? Because I think that just displaying "Show less" is fine?!

I also think that displaying just "Show less" is fine.

**Comment by akmaekki** [comment]
Created a pull request for this issue, refer https://github.com/HabitRPG/habitica/pull/11492. Is it possible for me to add labels to my pull request, like `status: pr: ready for review`?

**Comment by Alys** [comment]
@akmaekki it's not possible for you to add or change labels but we do it periodically. Thanks for asking though!

**Comment by shanaqui** [comment]
@akmaekki Hi! Are you still interested in working on this issue? Please let us know! :)

**Comment by akmaekki** [comment]
Hi, I'm still interested. But for me this problem was already solved. See the corresponding pull request.

**Comment by shanaqui** [comment]
Ah, I checked in because the pull request is marked as "needs work" -- is that in error, then? Or are there still changes required? @paglias

**Comment by paglias** [comment]
As mentioned in the PR the Stable page has some existing performance issues and the changes introduced are quite heavy in terms of computation and would need to be rewritten from Vue.js methods (re-rendered every time something changes) to computed properties that are re-rendered only when changed

**Comment by akmaekki** [comment]
I'll give it a try. Know about such problems from Angular :D

**Comment by shanaqui** [comment]
@akmaekki Hi! It's been a few weeks, so just checking in whether you're still interested in working on this or whether we should open it up for other contributors?

**Comment by shanaqui** [comment]
I'm marking this as help wanted again so anyone can pick it up. :)

**Comment by akmaekki** [comment]
@shanaqui 
If it's ok for you, I'll take a look into this again.
